torchvision/datasets/utils.py

```diff
@@ -249,3 +250,16 @@ def download_and_extract_archive(url, download_root, extract_root=None, filename
     archive = os.path.join(download_root, filename)
     print("Extracting {} to {}".format(archive, extract_root))
     extract_archive(archive, extract_root, remove_finished)
+
+
+def verify_str_arg(value, valid_values, arg):
+    if not isinstance(value, torch._six.string_classes):
+        msg = "Expected type str for argument {}, ".format(arg)
+        msg += "but got type {}.".format(type(value))
+        raise ValueError(msg)
+    # TODO: Should we call .lower() on value and valid_values?
```

Comment:
I think it's better to do exact string comparison, so not calling `.lower()` seems more robust to me.

If users want to have less strict checkings, they can do it on the dataset

Reply:
Should I keep it in `CelebA` for BC?

https://github.com/pytorch/vision/blob/d9830d8688bc2f2fc8070a4688d590c7c0686b27/torchvision/datasets/celeba.py#L69-L75

Reply:
Yeah, just add a `split.lower()` in `celeba.py`